src/xrc-tests/src/tests/can_successfully_cache_rates.rs

```diff
@@ -34,39 +34,51 @@ fn can_successfully_cache_rates() {
         },
     };
 
-    let responses = EXCHANGES
-        .iter()
-        .flat_map(|exchange| {
-            let json = get_sample_json_for_exchange(exchange);
-            [
-                build_crypto_exchange_response(
-                    exchange,
-                    &request.base_asset,
-                    timestamp,
-                    json.clone(),
-                ),
-                build_crypto_exchange_response(exchange, &request.quote_asset, timestamp, json),
-            ]
-        })
-        .collect::<Vec<_>>();
+    let responses = mock_responses::exchanges::build_responses(
+        request.base_asset.symbol.clone(),
+        timestamp,
+        |exchange| match exchange {
+            xrc::Exchange::Binance(_) => Some("41.96000000"),
+            xrc::Exchange::Coinbase(_) => Some("44.25"),
+            xrc::Exchange::KuCoin(_) => Some("44.833"),
+            xrc::Exchange::Okx(_) => Some("42.03"),
+            xrc::Exchange::GateIo(_) => Some("42.64"),
+            xrc::Exchange::Mexc(_) => Some("46.101"),
+            xrc::Exchange::Poloniex(_) => Some("46.022"),
+        },
+    )
+    .chain(mock_responses::exchanges::build_responses(
+        request.quote_asset.symbol.clone(),
+        timestamp,
+        |exchange| match exchange {
+            xrc::Exchange::Binance(_) => Some("41.96000000"),
+            xrc::Exchange::Coinbase(_) => Some("44.25"),
+            xrc::Exchange::KuCoin(_) => Some("44.833"),
+            xrc::Exchange::Okx(_) => Some("42.03"),
+            xrc::Exchange::GateIo(_) => Some("42.64"),
+            xrc::Exchange::Mexc(_) => Some("46.101"),
+            xrc::Exchange::Poloniex(_) => Some("46.022"),
+        },
+    ))
```

Comment:
Is there an elegant way to get rid of this code duplication?

Reply:
You could define a function that returns this function, right?

Reply:
Moved the entire setup to a common function.